actors/util/adt/balancetable_test.go

```diff
@@ -43,16 +49,69 @@ func TestBalanceTable(t *testing.T) {
 		assert.Equal(t, abi.NewTokenAmount(30), amount)
 	})
 
+	t.Run("Add works only if account has been created", func(t *testing.T) {
+		addr := tutil.NewIDAddr(t, 100)
+		bt := buildBalanceTable()
+
+		// fail if account hasnt been created
+		require.Error(t, bt.Add(addr, abi.NewTokenAmount(10)))
+
+		// now create account -> add should work
+		require.NoError(t, bt.AddCreate(addr, abi.NewTokenAmount(10)))
+		require.NoError(t, bt.Add(addr, abi.NewTokenAmount(5)))
+		bal, err := bt.Get(addr)
+		require.NoError(t, err)
+		require.Equal(t, abi.NewTokenAmount(15), bal)
+	})
+
+	t.Run("Get works only if account has been created", func(t *testing.T) {
+		addr := tutil.NewIDAddr(t, 100)
+		bt := buildBalanceTable()
+
+		// fail if account hasnt been created
+		bal, err := bt.Get(addr)
+		require.Error(t, err)
+		require.Equal(t, big.Zero(), bal)
+
+		// now create account -> add should work
+		require.NoError(t, bt.AddCreate(addr, abi.NewTokenAmount(10)))
+		bal, err = bt.Get(addr)
+		require.NoError(t, err)
+		require.Equal(t, abi.NewTokenAmount(10), bal)
+	})
+
+	t.Run("Must subtract fails if account not created or balance is insufficient", func(t *testing.T) {
+		addr := tutil.NewIDAddr(t, 100)
+		bt := buildBalanceTable()
+
+		// fail if account hasnt been created
+		require.Error(t, bt.MustSubtract(addr, abi.NewTokenAmount(0)))
+
+		// create account -> should work now
+		require.NoError(t, bt.AddCreate(addr, abi.NewTokenAmount(0)))
+		require.NoError(t, bt.MustSubtract(addr, abi.NewTokenAmount(0)))
+
+		require.NoError(t, bt.Add(addr, abi.NewTokenAmount(5)))
+		// error if insufficient balance -> however, balance is still subtracted
+		// TODO Is that correct ?
```

Comment:
You can link to this:  https://github.com/filecoin-project/specs-actors/issues/652.  Or if you resolve before merging you can close it down.